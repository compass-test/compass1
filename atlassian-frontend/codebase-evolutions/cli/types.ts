export type Flags = {
  readonly debug?: boolean;
};

export type ConfirmResponse = { answer: boolean };
export type InputResponse = { answer: string };
export type SelectResponse = { answer: string | string[] };

export type UserInputType = {
  task?: string;
  allDependencies?: string[];
  hasPostUpgradeTasks?: boolean;
  hasSingletons?: boolean;
  upgradeIdentifier?: string;
  rolloutStrategy?: string;
};

export type UserActionType = {
  key: string;
  question: string;
  answer: any;
};

export type Query = {
  // Used when asking the question, and in the summary
  question: string;
  // Only used when asking the question
  context?: string | string[];
};

export type Answer = string | string[] | boolean;

export type Choice = {
  value: string | string[];
  nextDecisionKey?: keyof UserInputType;
};

export type Question = {
  type: 'confirm' | 'select' | 'select-multi' | 'input';
  query: Query | string;
  // Choices presented when type is select
  choices?: Choice[];
};

export type Decision = {
  key: keyof UserInputType;
  question: Question;
  // User provided answer or choice
  answer?: Answer;
  nextKey: (this: Decision, answer: Answer) => keyof UserInputType | undefined;
  next?: () => Decision | undefined;
  prev?: () => Decision | undefined;
  setUserInput?: (userInput: UserInputType) => void;
};
