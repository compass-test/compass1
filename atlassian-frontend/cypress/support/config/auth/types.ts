export interface IdentityConfiguration {
  verifyHostname: string;
  loginHostName: string;
}

export interface CookieConfiguration {
  name: string;
  domain: string;
}

export interface AuthConfiguration {
  username: string;
  password: string;
  identity: IdentityConfiguration;
  cookie: CookieConfiguration;
}
