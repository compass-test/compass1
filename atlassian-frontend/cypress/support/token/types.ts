export interface IdentityVerifyResponse {
  token: string;
}
export interface TokenServiceOpts {
  identityHostname: string;
  identityLoginHostname: string;
  identityUsername: string;
  identityPassword: string;
  cookieName: string;
}

export interface UserSession {
  sessionToken: string;
  xsrfToken: string;
}
