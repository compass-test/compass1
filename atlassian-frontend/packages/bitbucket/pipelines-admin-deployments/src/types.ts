import { Environment } from '@atlassian/pipelines-models';

export type ReorderEnvironments = (
  type: string,
  environmentTypeOrder: { uuid: string }[],
) => void;

export type FetchBranches = (
  name: string,
) => Promise<{ values: { name: string }[] }>;

export type CreateBranchRestriction = (
  environmentUuid: string,
  pattern: string,
) => void;

export type DeleteBranchRestriction = (
  environmentUuid: string,
  restrictionUuid: string,
) => void;

export type CreateEnvironment = (
  environmentName: string,
  environmentType: string,
) => void;

export type UpdateEnvironment = (environmentUuid: string, change: any) => void;

export type DeleteEnvironment = (environmentUuid: string) => void;

export type RenderVariableList = (environment: Environment) => React.ReactNode;
