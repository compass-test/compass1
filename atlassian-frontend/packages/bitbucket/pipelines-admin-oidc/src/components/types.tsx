export type EnvironmentSelectItem = {
  label: string;
  value: string;
  uuid: string;
};

export type DeploymentEnvironment = {
  label: string;
  options: EnvironmentSelectItem[];
};
