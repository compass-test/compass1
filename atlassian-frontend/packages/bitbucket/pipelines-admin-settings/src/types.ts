export type Account = {
  hasFetchedUser: boolean;
  userHas2FaEnabled: boolean;
};

export type Capabilities = {
  hasFetchedCapabilities: boolean;
  pipelinesEnabled: boolean;
};

export type BuildConfiguration = {
  hasYmlFile: boolean;
  hasFetchedRawYmlFile: boolean;
};
