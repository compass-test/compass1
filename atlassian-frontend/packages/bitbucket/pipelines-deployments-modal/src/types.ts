export enum ModalType {
  PREVIEW = 'preview',
  SUMMARY = 'summary',
  REDEPLOY = 'redploy',
  STEP_SUMMARY = 'step_summary',
  STEP_PREVIEW = 'step_preview',
  STEP_REDEPLOY = 'step_redploy',
}

export type Line = {
  oldLine?: number;
  newLine?: number;
  type: 'normal' | 'add' | 'del' | 'empty' | 'loaded';
  content: string;
  wordDiff?: string;
  position?: number;
  conflictType: 'marker' | 'content';
};

export type ChunkEntry = {
  id: string;
  newStart: number;
  oldStart: number;
  newLines: number;
  oldLines: number;
  content: string;
  changes: Line[];
};

export type Diff = {
  chunks: ChunkEntry[];
  from: string;
  to: string;
};

export type DiffContent = {
  lineFrom?: number;
  lineTo?: number;
  content: string;
};

export type ChunkReducerAccumulator = {
  before: ChunkEntry;
  after: ChunkEntry;
};

export type ChangeReducerAccumulator = {
  before: Line[];
  after: Line[];
};

export type GroupChangeAccumulator = {
  loadedBefore: Line[];
  loadedAfter: Line[];
  rest: Line[];
};
