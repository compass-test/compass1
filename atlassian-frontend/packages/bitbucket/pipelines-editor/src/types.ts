import type { SupportedLanguages } from '@atlaskit/code/types';
import type { Variable } from '@atlassian/pipelines-variables';

export enum SyntaxType {
  Text,
  CodeBlock,
}

export interface PipeReadmeSyntax {
  type?: SyntaxType;
  html?: string;
  innerText?: string;
  language?: SupportedLanguages;
}

export type Step = {
  name: string;
  description: string;
  logo: any;
  yml: string;
};

export type Pipe = {
  category: string;
  name: string;
  description: string;
  logo: string;
  yml: string;
  repositoryPath?: string;
  version?: string;
  maintainer?: { name: string; website: string };
  vendor?: { name: string };
  timestamp?: number;
  tags: string[];
};

export type Template = {
  name: string;
  description: string;
  language: string;
  logo: string;
  yml: string;
};

export type VariablesProps = {
  createVariable: (variable: Variable, environmentUuid?: string) => void;
  deleteVariable: (variable: Variable, environmentUuid?: string) => void;
  updateVariable: (
    previousVariable: Variable,
    variable: Variable,
    environmentUuid?: string,
  ) => void;
  isFetchingVariables: boolean;
  isReadOnly: boolean;
  variables: Variable[];
  environments?: { [uuid: string]: string };
  environmentVariables?: { [uuid: string]: Variable[] };
};
