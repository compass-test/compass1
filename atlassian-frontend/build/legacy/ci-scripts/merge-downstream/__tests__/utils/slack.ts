import {
  sendFailedMessage,
  conflictedFilesBlock,
  constrainMessage,
} from '../../utils/slack';
import { MergeConflict, User } from '../../types';
import { SlackClient } from '@atlaskit/build-utils/slack';

const mockAxiosInstance = {
  get: jest.fn(),
  post: jest.fn(),
};
jest.mock('axios', () => ({
  create: jest.fn(() => mockAxiosInstance),
}));

const fakeClient = new SlackClient({
  channel: 'rm-channel',
  username: 'AFP Downstream Merge Notifier',
  token: 'token',
});

describe('merge downstream', () => {
  it('default message matches snapshot', async () => {
    mockAxiosInstance.post.mockResolvedValue({});
    mockAxiosInstance.get.mockResolvedValue({
      data: { ok: true, user: { id: 'rm-slack-id' } },
    });

    await sendFailedMessage(
      {
        from: 'from-branch',
        to: 'to-branch',
        releaseManager: {
          emailAddress: 'rm-email@atlassian.com',
        } as User,
        mergeBranch: 'merge-branch-name',
        pipelinesLink: 'pipeline-link',
      },
      [],
      fakeClient,
    );

    expect(mockAxiosInstance.post.mock.calls.length).toBe(1);
    expect(mockAxiosInstance.post.mock.calls[0][1].blocks).toMatchSnapshot();
  });

  describe('conflictedFilesBlock', () => {
    const textFieldCharacterThreshold = 3000;

    it('return a list of conflicted files as well as the person who last touched them', async () => {
      const conflict1: MergeConflict = {
        file: 'file1',
        authorName: 'Linus',
        authorEmail: 'lkarsai@atlassian.com',
      };
      const conflict2: MergeConflict = {
        file: 'file2',
        authorName: 'Roger',
        authorEmail: 'roger@atlassian.com',
      };

      const block = await conflictedFilesBlock(
        [conflict1, conflict2],
        fakeClient,
      );

      expect(block!.text!.text).toContain(conflict1.file);
      expect(block!.text!.text).toContain(conflict1.authorName);
      expect(block!.text!.text).toContain(conflict2.file);
      expect(block!.text!.text).toContain(conflict2.authorName);
    });

    it('only return first 20 files and display the proper text', async () => {
      const conflicts: Array<MergeConflict> = [];
      for (let i = 1; i <= 30; i++) {
        conflicts.push({
          file: `file-${i}`,
          authorName: 'Linus',
          authorEmail: 'lkarsai@atlassian.com',
        });
      }
      const block = await conflictedFilesBlock(conflicts, fakeClient);
      const message = constrainMessage(block!.text!.text);
      expect(message).toContain('file-1');
      expect(message).toContain('file-20');
      expect(message).not.toContain('file-21');
      expect(message).toContain('And 10 other files');
    });

    it(`returns a smaller text field message if the number of characters exceeds ${textFieldCharacterThreshold}`, async () => {
      const conflicts: Array<MergeConflict> = [];
      for (let i = 1; i <= 5; i++) {
        conflicts.push({
          file: `file-${i}_${'-'.repeat(textFieldCharacterThreshold)}`,
          authorName: 'foo',
          authorEmail: 'foo@atlassian.com',
        });
      }
      const block = await conflictedFilesBlock(conflicts, fakeClient);
      const message = constrainMessage(block!.text!.text);
      expect(message.length).toBeLessThan(textFieldCharacterThreshold);
      expect(message).toContain(
        'We are unable to display the conflicted files',
      );
    });

    it('return empty object if no conflicts', async () => {
      const block = await conflictedFilesBlock([], fakeClient);
      expect(block).toEqual({});
    });
  });
});
