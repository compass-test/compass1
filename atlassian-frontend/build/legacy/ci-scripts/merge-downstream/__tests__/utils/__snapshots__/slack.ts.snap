// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge downstream default message matches snapshot 1`] = `
Array [
  Object {
    "text": Object {
      "text": ":call_me_hand: *Calling release manager:* <@rm-slack-id>

:no_entry: *Automatic Merge Failed:* \`from-branch\` to \`to-branch\` :no_entry:

:hammer_and_wrench: <pipeline-link|See Pipeline for output>",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "type": "divider",
  },
  Object {},
  Object {
    "text": Object {
      "text": "*Instructions:*\`\`\`git fetch origin
git checkout -b merge-branch-name origin/to-branch
git merge origin/from-branch
# Fix all conflicts
git add <conflicted-filenames>
git commit --no-verify
# Uncomment \\"Conflicts:\\" section
git push -u origin merge-branch-name\`\`\`
Then, raise a PR from \`merge-branch-name\` to \`to-branch\`",
      "type": "mrkdwn",
    },
    "type": "section",
  },
  Object {
    "elements": Array [
      Object {
        "text": "For more information on this process, please see <http://go.atlassian.com/af-merging-master|go/af-merging-master>",
        "type": "mrkdwn",
      },
    ],
    "type": "context",
  },
  Object {
    "type": "divider",
  },
]
`;
